qadence/noise/protocols.py

```diff
@@ -149,15 +149,18 @@ def list(cls) -> list:
         return list(filter(lambda el: not el.startswith("__"), dir(cls)))
 
     def filter(self, protocol: NoiseEnum | str) -> NoiseHandler | None:
-        is_protocol: list = [p == protocol or isinstance(p, protocol) for p in self.protocol]  # type: ignore[arg-type]
-        return (
-            NoiseHandler(
+        is_protocol: list = list()
+        if protocol == NoiseProtocol.READOUT:
+            is_protocol = [p == protocol for p in self.protocol]
+        else:
+            is_protocol = [isinstance(p, protocol) for p in self.protocol]  # type: ignore[arg-type]
```

Comment:
Isn't possible to use the `filter` functionality ?

Reply:
Well `filter` would return the objects, but I need to know also the corresponding `options` for filtered protocols. Hence why I keep the list of booleans